src/canvasObject.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var CanvasAnimationHelpers = require('./CanvasAnimationHelpers');
+
+var CanvasObject = function(config) {
+  this.needed = config.needed || false;
+  this.visible = config.visible || true;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.position = config.position || { x: 0, y: 0 };
+  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
+
+  this.id = config.canvas['@id'];
+  this.height = config.canvas.height;
+  this.width = config.canvas.width;
+  this.images = config.canvas.images;
+  this.label = config.canvas.label;
+  this.tileSourceUrl = config.canvas.images[0].resource.service['@id'] + '/info.json';
+};
+
+CanvasObject.prototype = {
+  openTileSource: function(x, y, width, viewer) {
+    var self = this;
+
+    viewer.addTiledImage({
+      x: x,
+      y: y,
+      width: width,
+      tileSource: this.tileSourceUrl,
+      index: 0, // Add the new image below the stand-in.
+      success: function(event) {
+        var main = event.item;
+
+        var tileDrawnHandler = function(event) {
+          if (event.tiledImage === main) {
+            var previousImageObj = self.mainImageObj;
+
+            viewer.removeHandler('tile-drawn', tileDrawnHandler);
+            self._setMainImage(main);
+            main.setOpacity(0, true);
+            CanvasAnimationHelpers.fade(main, 1);
+
+            if(previousImageObj){
+              viewer.world.removeItem(previousImageObj);
+            }
+          }
+        };
+        viewer.addHandler('tile-drawn', tileDrawnHandler);
+      }
+    });
+  },
+
+  openThumbnail: function(x, y, width, viewer) {
```

Comment:
For that matter, maybe the x, y, width, viewer needed here should already have been passed in during the object constructor? 

The goal is for these objects to be as much in charge of their own destiny as possible, not relying on outside source of data (except at creation of course). 

On a related note, when we get into having alternate images and overlays, we'll probably want the CanvasObject to have its own idea of its bounds, independent of any of its tiled images. Kind of like how I did it here: 

https://github.com/openseadragon/openseadragon/blob/master/test/demo/m2/js/page.js